test_util/helpers.py

```diff
@@ -189,3 +248,15 @@ def remove_file():
     # Attempt to remove the file upon normal interpreter exit.
     atexit.register(remove_file)
     return temp_path
+
+
+def lazy_property(property_fn):
```

Comment:
@orsenthil we don't want a max size or anything (We want to always cache the lazy properties after first call) like that, and this isn't parameterized like lru_cache works.